### ISSUE TITLE
JSON form: numeric to px conversion

### ISSUE THREAD

**Issue by dvoytenko** [cristianbote/goober#179]
It could be good to have the JSON format more consistent with (P)react's `style` prop.

**Comment by cristianbote** [comment]
Indeed. That would be awesome to have, but I am worried about the added size. Wondering if it could be part of the `goober-autoprefixer` package instead 🤔  that way you'll get the conversion only if you used the autoprefixer.

**Comment by dvoytenko** [comment]
I think in Preact this is done with a rather [small footprint](https://github.com/preactjs/preact/blob/0c14a92a98e8ed8427aa4a9dc9db09045fc5b27a/src/diff/props.js#L36). However, even a separate plugin could make sense. However, if this was a core feature, it'd be safer with a component distribution since it won't rely on extra setup.

Overall, I'd say, this is a smaller priority. It's convenient for some cases to reuse some constants for both `style` and stylesheet via `css''`. But it's not a predominant case. I think the nice thing is that you could introduce this feature now or later and technically you'd not be breaking compatibility because, e.g. `top: 10` is currently not working technically, so if it starts working tomorrow it'd be a very soft compatibility bump. So as an option we could adopt Preact's code and see how much size it's adding?

**Comment by cristianbote** [comment]
Uh oh `+82B` added if I include the `IS_NON_DIMENSIONAL` rule from `preact`. Kind of a lot.

**Comment by dvoytenko** [comment]
In most of cases we just resolve stylesheets statically during compilation. But I think if the added size is a concern for you, I'd call it a lower priority.
